string data
string from_node
float32 timestamp